### PR TITLE
Clarify steps for adding an OIDC provider

### LINKED ISSUE #3952: OIDC configruation guide is not consistent on how to configure the clients
## Related issues
https://jira.camunda.com/browse/SUPPORT-22349

## Description
As part of the investigation into the linked support request, it was spotted that the docs relating to the configuration steps and setup for OIDC connections isn't consistent between `Generic` and `Microsoft Entra`.

To reduce confusion, and create a better experience, these docs should be aligned. The `Generic` tab produces the correct outcome which is that there is an app registration for each component in the Camunda stack that you wish to connect and not one app registration for the entire stack.

We should also take this opportunity to add a note in the `Component specific configuration` section that Web Modeler requires two clients, one for the Internal API and one for the External/Public API.

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description
Closes #3952 

The OIDC configuration instructions for MS Entra were misleading, and implied that you needed one application for every component. 

- Update the Entra ID instructions to clarify that each component needs its own application
- Clarify steps to be more explicit
- Update links to correct sections of MS docs
- Add note of two clients requirement for web modeler in component section
- Update use of heading from markdown to html to reduce broken links in the page TOC

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [X] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [X] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [X] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by conceptualshark** [comment]
I deployed this specifically so reviewers could check out the markdown > html update in action in the TOC:
https://preview.docs.camunda.cloud/pr-4150/docs/self-managed/setup/guides/connect-to-an-oidc-provider/
https://preview.docs.camunda.cloud/pr-4150/docs/next/self-managed/setup/guides/connect-to-an-oidc-provider/
https://preview.docs.camunda.cloud/pr-4150/docs/8.4/self-managed/platform-deployment/helm-kubernetes/guides/connect-to-an-oidc-provider/

Compare to live, where the TOC includes apparently duplicated links, and only the ones in the active tab work, which is a disruptive experience: https://docs.camunda.io/docs/next/self-managed/setup/guides/connect-to-an-oidc-provider/

File: docs/self-managed/setup/guides/connect-to-an-oidc-provider.md [hunk 1]
```diff
@@ -41,7 +37,7 @@ configuration](#component-specific-configuration) to ensure the components are c
 <Tabs groupId="authPlatform" defaultValue="generic" queryString values={[{label: 'Generic', value: 'generic' },{label: 'Microsoft Entra ID', value: 'microsoftEntraId' }]} >
 <TabItem value="generic">
 
-### Steps
+<h3>Steps</h3>
```
Comment (christinaausley) [inline docs/self-managed/setup/guides/connect-to-an-oidc-provider.md:40]: This is to avoid the complications across tabs, yes?

Comment (conceptualshark) [reply docs/self-managed/setup/guides/connect-to-an-oidc-provider.md:40]: Yeah, I figured this page was small enough that they weren't needed in the TOC, and it eliminates the weird duplication/linking.

File: docs/self-managed/setup/guides/connect-to-an-oidc-provider.md [hunk 2]
```diff
@@ -112,28 +108,27 @@ global:
...
-   of type `Single-page application` for Modeler. The expected redirect URI of the component you are configuring an app for can be found
-   in [component-specific configuration](#component-specific-configuration). Make sure the redirect URIs entered here match the redirect URI's configured in **Step 4**.
-
-3. Once you have registered a platform for your app a client secret needs to be created. To do this, see [adding a client secret](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app#add-a-client-secret). Make a note of the value of the client secret as it will be required later on.
+:::note
+Ensure you register a new application for each component.
+:::
 
-4. Set the following environment variables for the component you are configuring an app for:
+1. Within the Entra ID admin center, [register a new application](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app) for **each** component you would like to connect.
```
Comment (Ben-Sheppard) [inline docs/self-managed/setup/guides/connect-to-an-oidc-provider.md:124]: Just wondering, if we have this in here `for **each** component you would like to connect` do we need the note above? Maybe though this is a case of making this point very clear :)

**Review by Ben-Sheppard — APPROVED** [review]
Just one comment but its NABD for me :)

**Pull request merged**